primitives/rpc/src/lib.rs

```diff
@@ -177,8 +177,26 @@ sp_api::decl_runtime_apis! {
 		/// Return the elasticity multiplier.
 		fn elasticity() -> Option<Permill>;
 	}
+
+	#[api_version(2)]
+	pub trait ConvertTransactionRuntimeApi {
+		fn convert_transaction(transaction: ethereum::TransactionV2) -> <Block as BlockT>::Extrinsic;
```

Comment:
Consider changing this to with RLP `Vec<u8>` as input, and return `Option<Extrinsic>` where `None` indicates that the conversion failed.

Ethereum transaction's RLP encoding is backward and future compatible, but not really the SCALE encoding, as that's internal. Doing the above ensures that when runtime updates to support a new transaction type, you can immediately use it without waiting for old nodes to update. With this you can also remove the versioning.

Also, is there a particular reason that this isn't put into `EthereumRuntimeRPCApi`?

Reply:
Hmm I realized that our `EthereumRuntimeRPCApi` is not forward compatible anyway, so the above (using RLP `Vec<u8>` as input) may not provide much benefits.

The other points still stands -- maybe consider moving this to `EthereumRuntimeRPCApi`.

Reply:
We decided to have it in a separate trait so that the client can ask if the API is present or not.
Indeed, substrate does not allow to ask if a particular method of a runtime api is present or not.